virttest/env_process.py

```diff
@@ -1045,6 +1045,9 @@ def preprocess(test, params, env):
             kernel_extra_params_add += " intel_iommu=on"
         else:
             kernel_extra_params_remove += " intel_iommu=on"
+    guest_iommu_option = params.get("guest_iommu_option")
+    if guest_iommu_option:
+        kernel_extra_params_add += " iommu=%s" % guest_iommu_option
```

Comment:
I think it should work only if `enable_guest_iommu = yes`, do you?

Reply:
@luckyh I think they should be separated, when there is other options like `off`, `force`, `nobypass    [PPC/POWERNV]`, obviously it supports other platform besides Intel.

Reply:
@luckyh https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.1/html/installation_guide/appe-configuring_a_hypervisor_host_for_pci_passthrough :
```
If intel_iommu=on or amd_iommu=on works, you can try replacing them with iommu=pt or amd_iommu=pt. The pt option only enables IOMMU for devices used in passthrough and will provide better host performance. However, the option may not be supported on all hardware. Revert to previous option if the pt option doesn't work for your host.
```
Per this guide, even single `iommu=pt` without `intel_iommu=on`  could be a test scenario.

Reply:
@vivianQizhu  The info your provide is correct, `intel_iommu`, `amd_iommu`, `iommu=X` can work with bare mental.   Here is about guest kernel line,  QEMU only support `-device intel-iommu` now.

Reply:
@yanan-fu Yes, so we do not consider `amd_iommu`. My point is, if we consider all `iommu=` options, we are considering it as an independent kernel option, then it should not be bound to `intel_iommu=on`.

Reply:
@vivianQizhu Yes, `iommu=*` can work without `intel_iommu=on` (no meaning, but can work) in guest kernel line as an independent kernel option.  My point here is,  QEMU only support `intel_iommu`,  this option only work with `utils_misc.get_cpu_vendor(verbose=False) == 'GenuineIntel'`

Reply:
@yanan-fu So no modification required from you to my current code, right?

Reply:
@vivianQizhu No,  let me be more clear. Current code don't check the cpu vendor, but QEMU only support `intel-iommu` device now. We need add this step.

Reply:
@vivianQizhu ah...  please ignore my comment above, i only consider x86(only intel, no amd), miss other arch, such as ppc. It is not suitable.

Reply:
@yanan-fu Sorry, I don't agree, qemu command line only support intel has nothing to do with this `iommu` kernel option, that should not be controlled here.

Reply:
@vivianQizhu  Without iommu device, it is meaningless to add `iommu=*` option in kernel line. Both for host (hardware iommu) and guest ( - device intel-iommu).

Reply:
@vivianQizhu  For `enable_guest_iommu`, we check the vendor, because VM can only have `intel-iommu` device, so we can only add `intel-iommu=on` in guest kernel line. (In fact, if only consider kernel option, both intel_iommu and amd_iommu are support).
For iommu opyion, i think they are same.

Reply:
@yanan-fu I understand your point, but `iommu` itself does not distinguish that, here is not the right place the restrict it, and `intel_iommu` is restrict to intel because the option itself only support intel. Hard to say we don't support other platform in the further. Actually, even qemu guest also works with Amd iommu, we have such product bz like amd should not work with iommu device but devel said it make sense since code are actually same.
If you want to make sure kernel options only added when there is iommu device, then should check if there is such device, not check platform. But still does not make sense to me. Cause the kernel options are independent from the hardware device, they should be configured by user, should not restrict the kernel cmdline when there is no such device.
Anyway, it should be configured in right way, rather than restrict here, which is logically wrong for me.